rpcs3/Emu/RSX/VK/VKGSRender.cpp

```diff
@@ -3373,6 +3373,46 @@ void VKGSRender::flip(int buffer, bool emu_flip)
 		vk::change_image_layout(*m_current_command_buffer, target_image, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, present_layout, range);
 	}
 
+	if (m_frame->screenshot_toggle == true)
+	{
+		m_frame->screenshot_toggle = false;
+
+		const size_t sshot_size = buffer_height * buffer_width * 4;
+
+		const auto device = vk::get_current_renderer();
+		vk::buffer sshot_vkbuf(
+		    *device, 256 * 0x100000, device->get_memory_mapping().host_visible_coherent, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT, VK_BUFFER_USAGE_TRANSFER_SRC_BIT, 0);
```

Comment:
Instead of hardcoding this, just align sshot_size up to the next megabyte. A helper method is already provided, just use `align(sshot_size, 0x100000)`.

Reply:
Buffer usage should be TRANSFER_DST here, not TRANSFER_SRC. The transfer pipe only writes to it via CopyBufferToImage. Note that map/unmap are host access and need not be explicitly described here.